freshmaker/lightblue.py

```diff
@@ -1825,3 +1830,50 @@ def get_fixed_published_image(self, name, version, image_group, rpm_nvrs, conten
         image = images[0]
         image.resolve(self)
         return image
+
+    def is_latest_eus_image(self, image):
+        """
+        Check if the image is EUS base image and his repository marked with
+        auto_rebuild tag. EUS repository contains '-els' suffix.
+
+        :param ContainerImage image: image to check
+        :param dict repo_to_auto_rebuild_tags: mapping of processed repositories
+            names to set of their 'auto_rebuild_tags'
+        :return: True if image is EUS base image with 'auto_rebuild_tag', False otherwise
+        """
+        def get_repo(repository):
+            query = {
+                "objectType": "containerRepository",
+                "query": {
+                    "$and": [
+                        {
+                            "field": "repository",
+                            "op": "=",
+                            "rvalue": repository
+                        },
+                    ]
+                },
+                "projection": [
+                    {"field": "auto_rebuild_tags", "include": True, "recursive": True}
+                ]
+            }
+            repos = self.find_container_repositories(query, auto_rebuild=True)
+            return repos[0]
+
+        for repo in image["repositories"]:
+            repo_name = repo["repository"]
+            # Check tags only if it's EUS image
+            if re.match(r".*-els($|\W)", repo_name):
```

Comment:
How about a more precise reged?
```suggestion
            if re.match(r"rhel\d+-\d+-els\/rhel", repo_name):
```

Reply:
But if we will have not only RHEL EUS images? And in design doc there is this example of repository: `rh-osbs/rhel-els`. It will not match the regex.

Reply:
@apaplaus so it's only the RHEL EUS image that isn't published and needs rebuilding as far as I know. The `rh-osbs/rhel-els` repo is the "build repo", which is the repo that is used by OSBS to store built images in quay.io. Any image that is released then gets copied from the build repo to the Red Hat Container Registry. This means we specifically want to ignore the `rh-osbs` namespace since autorebuild tags will never be set on that.

Reply:
I've changed it